saleor/graphql/checkout/tests/benchmark/test_checkout_mutations.py

```diff
@@ -912,9 +912,10 @@ def test_add_checkout_lines_with_external_shipping(
         api_client.post_graphql(MUTATION_CHECKOUT_LINES_ADD, variables)
     )
     assert not response["data"]["checkoutLinesAdd"]["errors"]
-    # One api call:
+    # Two api calls:
     # - post-mutate() logic used to validate currently selected method
-    assert mock_send_request.call_count == 1
+    # - fetch_checkout_prices_if_expired - calculating all prices for checkout
+    assert mock_send_request.call_count == 2
```

Comment:
Could we add checks for these calls?

Reply:
Are we sure that we should have duplicated external calls in this mutation?

Reply:
Full background here:

Those are actually duplicated calls and their number was already 2 in this optimization pull request.
https://github.com/saleor/saleor/pull/8744/files#diff-732cfdb4bfc0be8786c912bc299bb2ea95e56326ca5c8d62907c82117886e057R915-R917

In a subsequent pull request calls to external shipping methods have been wrapped with lazy initialization which reduced the number of calls to 1, since the shipping method wasn't queried in particular benchmark test. The reason for the number of calls on this pull request going back to 2 is that `fetch_checkout_prices_if_expired` requires evaluating the shipping price.

The real reason for the duplicated call here is the fact that `ChecokutInfo` is evaluated twice:
- mutations call `fetch_checkout_info`
- `checkout_info.checkout` is then passed as the root object to the GraphQL checkout type
- finally another `CheckoutInfo` instance is constructed inside of `CheckoutInfoByCheckoutTokenLoader`

There are 2 possible solutions:
- add cache for external shipping methods (there's a draft PR somewhere that addresses this)
- use CheckoutInfo as the root object for Checkout queries, as this will avoid double evaluation in all checkout mutations